### PR TITLE
Use the shared GitHub Actions workflows for buildpack releases

### PR DESCRIPTION
**Pull request by colincasey**
This sets up the release automation scripts from [languages-github-actions](https://github.com/heroku/languages-github-actions) for the Python buildpack.

Also adds the target Docker repository to `buildpack.toml` to align with the metadata used by other Heroku buildpacks.

GUS-W-13658750.

### CODE REVIEW

**Comment by edmorley** [comment]
Thank you for working on this btw! I'll have another review pass once I've had a chance to test some more, and once we've triaged some more of https://github.com/heroku/languages-github-actions/issues :-)

**Review by edmorley — CHANGES_REQUESTED** [review]
Marking changes requested to make the current state clearer.

This is blocked on:
- the review comments on the PR
- having a few more of the issues reported at https://github.com/heroku/languages-github-actions/issues fixed first

...but I'm very much looking forward to this landing! :-)

File: .github/workflows/release.yml
Comment (colincasey) [reply .github/workflows/release.yml:11]: each workflow is parameterized to work with a single buildpack if you give it the correct metadata so, technically, you could drop this job but then you'd have to replace it with another that did roughly the same thing so i doubt it would be worth it.

also, for consistency with other buildpacks i think it makes sense that each project maintains the same workflow.

File: .github/workflows/prepare-release.yml
Comment (colincasey) [reply .github/workflows/prepare-release.yml:18]: i'll have to test to see.  it was needed early in development before the workflows were extracted into reusable parts.

Comment (colincasey) [reply .github/workflows/prepare-release.yml:18]: @edmorley after some testing we only need to set `contents: write` which is required by the github release action.  

we could also drop the `contents: write` permission by switching to using the GH App to perform GitHub Release actions but that would require all the existing workflows to supply those credentials.  do you think that would be better?

File: .github/workflows/prepare-release.yml
Comment (colincasey) [reply .github/workflows/prepare-release.yml]: you could either:
- run the prepare release workflow and review the PR it creates.  this can be closed if you see an issue or are just testing
- download the shared actions repository, build the `actions` binary locally, then run the `prepare-release` command manually in the directory of your repository

File: buildpack.toml
Comment (edmorley) [inline buildpack.toml:23]: Reading the TOML spec now (I always forget the TOML table nuanced), it seems that intermediate tables can be omitted when defining a child table. ie: It should be possible to omit eg the bare `[metadata]` table if we wanted:
https://toml.io/en/v1.0.0#table

Also, it's possible to write tables inline to make things less verbose:
https://toml.io/en/v1.0.0#inline-table

Combining those two, this should be equivalent:

```suggestion
[metadata.release]
image = { repository = "docker.io/heroku/buildpack-python" }
```

...and I think is a bit more concise - plus if we were ever to add more `metadata.release` options, they could be neatly grouped under `[metadata.release]` rather than needing a new `[metadata.release.<FOO>]` table for every new config option.

File: .github/workflows/release.yml
Comment (edmorley) [inline .github/workflows/release.yml:1]: (there is just the one buildpack for now)

```suggestion
name: Release Buildpack
```

File: .github/workflows/prepare-release.yml
Comment (edmorley) [inline .github/workflows/prepare-release.yml:1]: (there is just the one buildpack for now)
```suggestion
name: Prepare Buildpack Release
```

File: .github/workflows/prepare-release.yml
Comment (edmorley) [inline .github/workflows/prepare-release.yml:9]: (for consistency with the quotes used for the `description` field a few lines higher, plus matches the Prettier output style)

```suggestion
        default: "patch"
```

**Pull request merged**
